src/Microsoft.VisualStudio.Composition/ByValueEquality+AssemblyNameComparer.cs

```diff
@@ -32,8 +40,8 @@ public bool Equals(AssemblyName x, AssemblyName y)
                     return true;
                 }
 
-                // fast path
-                if (x.CodeBase == y.CodeBase)
+                // If fast check is enabled, we can compare the code bases
+                if (this.fastCheck && x.CodeBase == y.CodeBase)
```

Comment:
It's used as a comparer for some dictionaries (e.g input assemblies). I was hesitant to remove the fast check before knowing the potential perf impact (I'm trying to get a val build with these changes in but it's proving to be a bit difficult - I can't uses the vs-mef build definition since my branch does not exist in this repo).